### ISSUE TITLE
SYN_SENT problem AP got disconnected one by one

### ISSUE THREAD

**Issue by eelco2k** [jacobalberty/unifi-docker#196]
### Host operating system

`docker` on Synology

### What tag are you using

`stable`

Problem:

I'm trying to use Unifi Controller in Docker via Synology. I've setup all correct port forwardings into my firewall. Furthermore i'm using nginx reverse proxy for management. but not for the :8080/inform (this one goes directly through my Synology --> Docker -> Unifi Controller Container.

I did a succesful migration/adoptation to change the inform URL and also checked "override inform host"

In the management UI they are all connected. But after some time one got disconnected. And within one day 6 got disconnected. 

At first i thought NAT on the client side was the problem as my NAT sessions table showed:

tcp SYN_SENT to my unifi.(myhost.com):8080/inform  

After i restarted my (docker container) all AP came online again (provisioning first and then online) But again after some time one missed the heartbeat and now disconnected again.

here is some part of the server log, i've removed mac addresses and changed to 00,01 and 02:

```
<webapi-623> WARN  api    - websocket session error:Unable to unwrap data, invalid status [CLOSED]
<webapi-626> WARN  api    - websocket session error:Unable to unwrap data, invalid status [CLOSED]
<inform-6> INFO  inform - from [00:00:00:00:00:01](, U7PG2, 4.0.21.9965): state=CONNECTED, last_inform=54, ext/stun_ip=172.17.0.1, dev_ip=192.168.1.14, up=9901
<devmgr> INFO  event  - [event] AP[00:00:00:00:00:00] was disconnected
<devmgr> INFO  dev    -    [state] dev[00:00:00:00:00:00] HEARTBEAT_MISSED->UNKNOWN, state_expire=0
<inform-6> INFO  inform - from [00:00:00:00:00:02](, U7PG2, 4.0.21.9965): state=CONNECTED, last_inform=55, ext/stun_ip=172.17.0.1, dev_ip=192.168.1.12, up=10204

INFO  api    - [api] api.err.InvalidVerification: /api/login
<webapi-679> INFO  api    - [api] api.err.InvalidVerification: /api/login
<webapi-682> INFO  api    - [api] api.err.InvalidVerification: /api/login
<webapi-671> INFO  api    - [api] api.err.InvalidVerification: /api/login
<webapi-677> INFO  api    - [api] api.err.InvalidVerification: /api/login
<webapi-658> INFO  api    - [api] api.err.InvalidVerification: /api/login
```

When i SSH into one of my accesspoints this is the info and log:

```
Model:       UAP-AC-Pro-Gen2
Version:     4.0.21.9965
MAC Address: 00:00:00:00:00:00
IP Address:  192.168.1.17
Hostname:   AC-pro
Uptime:      12519 seconds

Status:      Unknown[11] (http://unifi.(myhost.com):8080/inform)
```

and logs:
```
Tue Apr 30 13:39:38 2019 user.err : ace_reporter.reporter_fail(): Unknown[11] (http://unifi.(myhost.com):8080/inform)
Tue Apr 30 13:39:38 2019 user.err : ace_reporter.reporter_fail(): initial contact failed #4, url=http://unifi.(myhost.com):8080/inform, rc=11
Tue Apr 30 13:40:23 2019 user.err : ace_reporter.reporter_fail(): Unknown[11] (http://unifi.(myhost.com):8080/inform)
Tue Apr 30 13:40:23 2019 user.err : ace_reporter.reporter_fail(): initial contact failed #1, url=http://unifi.(myhost.com):8080/inform, rc=11
```
I can resolve my domainname from the AP in ssh.

but when i do:
``
curl http://unifi.(myhost.com):8080/inform -v
``
i'm not getting any response.

serverlogs:
```
<inform-6> ERROR inform - Inform Invalid for Device[172.17.0.1:49929], Content too short    <-- my CURL i guess

<devmgr> INFO  event  - [event] AP[00:00:00:00:00:00] was disconnected
[2019-04-30T13:39:14,094] <devmgr> INFO  dev    -    [state] dev[00:00:00:00:00:00] HEARTBEAT_MISSED->UNKNOWN, state_expire=0
<inform-6> INFO  dev    -    [state] dev[00:00:00:00:00:00] UNKNOWN->CONNECTED, state_expire=0
[2019-04-30T13:41:24,919] <inform-6> INFO  event  - [event] AP[00:00:00:00:00:00] was connected
INFO  inform - <<< [setparam] dev[00:00:00:00:00:00]: [_type, blocked_sta, server_time_in_utc]
<devmgr> INFO  dev    -    [state] dev[00:00:00:00:00:00] CONNECTED->HEARTBEAT_MISSED, state_expire=0

```
After 30 seconds or so i get back from cURL:
``
curl: (7) Failed to connect to unifi.(myhost.com) port 8080: Operation timed out
``
All the other AP's are connected but will fallout after each other. Restart of Docker is necessary to have the AP's up again.

Hope somebody can shine a light on this weird issue.

**Comment by alexanderharm** [comment]
I have exactly the same problem. Devices disconnect at random with error 11.

**Comment by alexanderharm** [comment]
I tried as alternative the docker image <https://hub.docker.com/r/goofball222/unifi> and since then I didn't have a single disconnect. Maybe one could compare the differences to figure out the issue with this image.

**Comment by jacobalberty** [comment]
@alexanderharm there really aren't any significant differences between the two aside from alpine vs ubuntu, and afaik the supported configuration from ubiquiti is on an os like ubuntu/debian. What configuration for the docker side of things did you use? and did you change it when switching to goofball222

**Comment by alexanderharm** [comment]
@jacobalberty No, I did not change anything in the Docker setup. However, goofball222 sets some JVM environmental parameters. Maybe that's a path to investigate further?

**Comment by jacobalberty** [comment]
@alexanderharm do a `ps auxwww | grep java`
You should see something like 
```
`java -Dunifi.datadir=/unifi/data -Dunifi.logdir=/unifi/log -Dunifi.rundir=/var/run/unifi -Xmx1024M -Djava.awt.headless=true -Dfile.encoding=UTF-8 -jar /usr/lib/unifi/lib/ace.jar start`
```

That -Xmx1024M is the JVM param goofball222 sets, that ps comes from this image though, if there are any other differences i'd be interested to know though

**Comment by alexanderharm** [comment]
The interesting thing is that it only happens to remote sites. The UniFi devices in the same network as the Docker host work just fine.

**Comment by alexanderharm** [comment]
@jacobalberty We tested now Goofball's for more than a month and we did not have a single issue with disconnecting devices. So something is different whatever it might be...

**Comment by jacobalberty** [comment]
@alexanderharm could you do `ps auxwww | grep java`  and see if theres a difference in how java is being called in that image? under my image its `java -Dunifi.datadir=/unifi/data -Dunifi.logdir=/unifi/log -Dunifi.rundir=/var/run/unifi -Xmx1024M -Djava.awt.headless=true -Dfile.encoding=UTF-8 -jar /usr/lib/unifi/lib/ace.jar start` im still interested to know if goofball's runs differently.

**Comment by alexanderharm** [comment]
On goofball's it is

```
unifi       72  0.6  3.6 4787252 296572 pts/0  Sl+  Dec01  38:38 /usr/bin/java -Xmx1024M -Dunifi.datadir=/usr/lib/unifi/data -Dunifi.logdir=/usr/lib/unifi/logs -Dunifi.rundir=/usr/lib/unifi/run -Djava.awt.headless=true -Dfile.encoding=UTF-8 -jar /usr/lib/unifi/lib/ace.jar start
```

It is the same, no?

**Comment by alexanderharm** [comment]
Maybe the Docker container reports the wrong IP (of the Docker network instead the host)? Saw that somewhere in the issues of another UniFi container...

**Comment by daballiemo** [comment]
I have the same behaviour. Remote site disconnect after a the container has been running for about 28 hours. Rebooting the container brings everything online again. There is no loss of connectivity on the remote sites. Local site stays up without a hitch.

Config:

docker run  --init \
-p 8080:8080 \
-p 8843:8443 \
-p 3478:3478/udp \
-p 10001:10001/udp \
-e TZ='Europe/Amsterdam' \
-v /volume1/docker/unifi:/unifi \
--name unifi jacobalberty/unifi:stable

Running on Synology DS918+, 12Gb mem and latest DSM version. 

Currently my workaround is to restart the container every night.

**Comment by GuillaumeBolo** [comment]
I don't no if it's SYN_SENT issue, but i have the same disconnection issue for remotes sites after 1 or 2 days. Not all devices in same time but one by one over time. 

Same workaround for a year : restart container every 24h 

DS415+ with last Docker update, using unifi:stable updated regularly.
1 local site
2 remotes sites

Thanks for any information.
